### ISSUE TITLE
not starting

### ISSUE THREAD

**Issue by TIBTHINK** [MartyMcKnife/BeatSaberSpotify#9]
so I type in all the variables for the prompt but the program still doesn't run, I'm not sure whats wrong with it.
there's is no log, no messages I'm trying to get it working for the oculus quest 2 but nothing

**Comment by MartyMcKnife** [comment]
The most likely reason for this issue is that you haven't installed the latest version of Python (v 3.9). To check this, open up cmd, and type 
`python --version`

If you don't have python 3.9 installed, you can get it [here](https://www.python.org/downloads/release/python-391/) 

If you do have python 3.9 installed, make sure 
`C:\Users\%USERPROFILE%\AppData\Local\Programs\Python\Python39\`
is a path

Failing this, I might have to do some more digging

**Comment by TIBTHINK** [comment]
it worked but, is there any reason that it has to be 3.9
normal 3.6 - 3.9 code works fine 

side note it crashed saying list index out of range
[beatsaberspotify.log](https://github.com/MartyMcKnife/BeatSaberSpotify/files/5693377/beatsaberspotify.log)

**Comment by MartyMcKnife** [comment]
~~99.9% of the code will work for Python 3. However, Python 3.9 introduced encoding= to the logging.basicConfig~~
For whatever reason I was still remembering my old code, which used the above. You are completely correct, 3.6-3.9 should work

As for the error, can you give me the spotify uri you were using? (This should've appeared in the log file, but there was another bug stopping this). The reason this is happening is because it can't find the URL for the covers, but I need to see whether that's me not catching the error, or an underlying problem with the script

**Comment by TIBTHINK** [comment]
https://open.spotify.com/playlist/1EjFTfO05q320UzMst8JJX?si=6ypL81jkRxORSyPYkmaaow

if you need help with the code dont be afraid to ask

**Comment by MartyMcKnife** [comment]
As I thought, it was a bug where local songs didn't have a cover URL, so the program errored out. I've fixed this for the latest version, which should come out sometime tomorrow - just fixing up some more bugs and additions

If you don't want to wait till then, just replace the write_tracks function in spotify.py with this:

```python
    def write_tracks(self, text_file, tracks, pathForCovers):
        with open(text_file, 'w+', encoding='utf-16') as file_out:
            i = 0
            while True:
                for item in tracks['items']:
                    i+=1
                    if 'track' in item:
                        track = item['track']
                    else:
                        track = item
                    
                    try:
                        if "Tracks" in text_file:
                            track_url = track['name']
                            for char in self.invalid:
                                track_url = track_url.replace(char, '')
                            file_out.write(track_url + '\n')
                        # Write Artists to Text File
                        elif "Artists" in text_file:
                            artists = track['artists']
                            artistsName = artists[0]['name']
                            for char in self.invalid:
                                artistsName = artistsName.replace(char, '')
                            file_out.write(artistsName + '\n')
                        #Download Album Cover
                        if pathForCovers != False:
                            if i == 1:
                                print('Downloading Album covers', flush=True)
                            images = track["album"]["images"]
                            url = images[0]["url"]
                            filename = track["name"]

                            for char in self.invalid:
                                filename = filename.replace(char, '')
                            path_full = os.path.join(pathForCovers, filename + str(".png"))
                            if not os.path.isfile(path_full):
                                wget.download(url, path_full)
                    except (KeyError, IndexError):
                        print(u'Skipping track {0} by {1} (local only?)'.format(
                                track['name'], track['artists'][0]['name']), flush=True)
                    

                # 1 page = 50 results
                # check if there are more pages
                if tracks['next']:
                    tracks = self.spotify.next(tracks)
                else:
                    break
```

**Comment by MartyMcKnife** [comment]
Fixed with v2.1
